app/forms/personal_details_form.rb

```diff
@@ -0,0 +1,113 @@
+class PersonalDetailsForm < Form
+  NAME_REGEX_FILTER = /\A[^"=$%#&*+\/\\()@?!<>0-9]*\z/
+  NINO_REGEX_FILTER = /\A[A-Z]{2}[0-9]{6}[A-D]{1}\Z/
+  DOB_PARAM_CONVERSION = {
+    "date_of_birth(3i)" => "day",
+    "date_of_birth(2i)" => "month",
+    "date_of_birth(1i)" => "year"
+  }
+
+  attribute :first_name
+  attribute :middle_name
+  attribute :surname
+  attribute :day
+  attribute :month
+  attribute :year
+  attribute :date_of_birth
+  attribute :national_insurance_number
+
+  validates :first_name, presence: {message: "Enter your first name"}
+  validates :first_name,
+    length: {maximum: 100, message: "First name must be less than 100 characters"},
+    format: {with: NAME_REGEX_FILTER, message: "First name cannot contain special characters"},
```

Comment:
I think this is fine as is but might be tempted to pull out Ana activemodel validator for this REGEX. Not a blocker though!